### PR TITLE
DRIVERS-1993: Convert ADL spec tests to unified test format 

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/DRIVERS-1993

- [x] Update changelog.
- [x] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver.
- [x] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

POC: https://github.com/mongodb/mongo-php-library/pull/1250

### CODE REVIEW

File: source/atlas-data-lake-testing/tests/unified/getMore.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                sort: { a: 1 }
+                batchSize: 3
+                limit: 4
+              commandName: find
+              databaseName: *database0Name
+          - commandStartedEvent:
+              command:
+                getMore: { $$type: [ int, long ] }
+                # collection name will be an internal identifier
+                collection: { $$type: string }
```
Comment (jmikola) [inline source/atlas-data-lake-testing/tests/unified/getMore.yml:44]: The original tests didn't assert `getMore` or `collection` at all, but we can now use `$$type`.

**Review by jyemin — APPROVED** [review]
LGTM

**Review by alcaeus — APPROVED** [review]

**Comment by jmikola** [comment]
Note: the failed link check on https://github.com/10gen/mongohouse/blob/master/README.md is a false positive, as the file exists in a private repository.

**Comment by jmikola** [comment]
Link check will be fixed in https://github.com/mongodb/specifications/pull/1549

I didn't realize that was a permanent lint failure.

**Pull request merged**